packages/flutter/lib/src/material/tabs.dart

```diff
@@ -400,9 +405,33 @@ class TabBarSelection {
     _previousIndex = _index;
     _index = value;
     _indexIsChanging = true;
+
+    // If the selected index change was triggered by a drag gesture, the current
+    // value of _performance.progress will reflect where the gesture ended. While
+    // the drag was underway progress indicates where the indicator and TabBarView
+    // scrollPosition are vis the indices of the two tabs adjacent to the selected
+    // one. So 0.5 means the drag didn't move at all, 0.0 means the drag extended
+    // to the beginning of the tab on the left and 1.0 likewise for the tab on the
+    // right. That is unless the selected index was 0 or maxIndex. In those cases
+    // progress just moves between the selected tab and the adjacent one.
+    // Convert progress to reflect the fact that we're now moving between (just)
+    // the previous and current selection index.
+
+    double progress;
+    if (_performance.status == PerformanceStatus.completed)
+      progress = 0.0;
+    else if (_previousIndex == 0)
+      progress = _performance.progress;
+    else if (_previousIndex == maxIndex)
+      progress = 1.0 - _performance.progress;
+    else if (_previousIndex < _index)
+      progress = (_performance.progress - 0.5) * 2.0;
+    else
+      progress = 1.0 - _performance.progress * 2.0;
```

Comment:
I think you're right. I added some asserts to verify that two or more tab labels or views are specified, since that's what's supported now.  Filed an issue for the general case:
https://github.com/flutter/flutter/issues/970